### ISSUE TITLE
const_eval interprets comparison-result as integer

### ISSUE THREAD

**Issue by oli-obk** [rust-lang/rust#25307]
`TEST2` is a constant int with the value `1`
```rust
    const X: i32 = 3;
    const TEST3: bool = X < 5;
    const TEST2: bool = TEST3;
```

`TEST5` is also a constant int with the value `1`
```rust
    const Y: f64 = 3.14;
    const TEST4: bool = Y < 5.1;
    const TEST5: bool = TEST4;
```

I don't think one can notice this outside of compiler plugins (I found it in https://github.com/oli-obk/rust-sa/blob/master/src/lib.rs#L26-34)

**Comment by oli-obk** [comment]
cc #23897

**Comment by oli-obk** [comment]
has been fixed in #29797
